### PR TITLE
Reverting #1042 and adding shape assertions to the MixtureDistribution

### PR DESCRIPTION
**Pull request by PascalIversen**
*Issue #, if available:*
 #1041

*Description of changes:*
Partly, reverting the changes of #1042 of the `MixtureDistribution` sampling. As I explained as a comment to the issue  #1041 there was no bug. Nevertheless, the shape of `mixture_probs` can be a source of confusion, thus I added shape assertions.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/gluonts/mx/distribution/mixture.py [hunk 1]
```diff
@@ -60,6 +61,32 @@ def __init__(
...
+            # assert that mixture_probs has the right shape
+            assertion_message = f"""mixture_probs have shape {mixture_probs.shape}, but expected shape: (..., k), 
+                                    where k is len(components)={len(components)}. 
+                                    All axes except the last one should either coincide with the ones from the 
+                                    component distributions, 
+                                    or be 1 (in which case, the mixing coefficient is shared across
+                                    the axis)."""
+
+            expected_shape = self.batch_shape + (len(components),)
+            assert len(expected_shape) == len(mixture_probs.shape), (
```
Comment (lostella) [inline src/gluonts/mx/distribution/mixture.py:80]: How come the test cases added in #1042 are not failing here?

**Review by lostella — APPROVED** [review]
Looks good!

File: src/gluonts/mx/distribution/mixture.py (hunk 1 above)
Comment (PascalIversen) [reply src/gluonts/mx/distribution/mixture.py:80]: They are not really probing the edge case:

```python
p=mx.nd.array([0.2]),
mixture_probs = mx.nd.stack(p, 1.0 - p, axis=-1)
mixture_probs.shape
(1,2)
```
=> The shape of the added test cases is correct.
